datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -250,7 +250,12 @@ impl DisplayAs for SortMergeJoinExec {
                 write!(
                     f,
                     "SortMergeJoin: join_type={:?}, on=[{}]{}",
-                    self.join_type, on, display_filter
+                    self.join_type,
+                    on,
+                    self.filter
+                        .as_ref()
+                        .map(|f| format!(", filter={}", f.expression()))
+                        .unwrap_or("".to_string())
```

Comment:
Why moving the code?

Reply:
inlined the display filter and changed `map_or_else` to map with default